### PR TITLE
[ZEPPELIN-5830] Add Zeppelin Security page to the website

### PR DESCRIPTION
**Pull request by raboof**
### What is this PR for?

This PR adds a 'Security' page to the website explaining what security characteristics people can expect from Zeppelin, and how to report any security issues people might find.

### What type of PR is it?
Improvement
Documentation

### What is the Jira issue?

[ZEPPELIN-5830](https://issues.apache.org/jira/browse/ZEPPELIN-5830)

### How should this be tested?

I had trouble running jekyll on my machine, so I have not verified the link in the menu and the page itself look good when rendered yet.

### Questions:
* Does the licenses files need to update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: security.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+## Authentication
+
+If you expose your Zeppelin instance on a network you don't fully trust,
+you should configure [Apache Shiro authentication](https://zeppelin.apache.org/docs/latest/setup/security/shiro_authentication.html).
+
+Non-authenticated users cannot view, store or execute notes, so they
+cannot execute code on the server or on other users' browsers.
+Authenticated users, however, have the same access as described above,
+so even when using authentication it is still important to only give
+trusted users access to Zeppelin. Specifically, users have access to
```
Comment (Reamer) [inline security.md:54]: Owner of Notes can set permissions on it. Therefore, an authenticated user does not see all Notes of all other users.

Comment (Reamer) [reply security.md]: Resolved via a personal Slack communication.

**Comment by zjffdu** [comment]
Thanks @raboof , PR LGTM

**Comment by Reamer** [comment]
@zjffdu 
I am not familiar with the homepage. Please merge the PullRequest if everything fits.

**Review by epugh — COMMENTED** [review]
content looked good to me!

**Review by zjffdu — APPROVED** [review]

**Pull request merged**